builtInFunctions/setGuardian.go

```diff
@@ -41,20 +43,36 @@ func (sg *setGuardian) ProcessBuiltinFunction(
 	acntSnd, _ vmcommon.UserAccountHandler,
 	vmInput *vmcommon.ContractCallInput,
 ) (*vmcommon.VMOutput, error) {
+	if check.IfNil(acntSnd) {
+		return nil, fmt.Errorf("%w for sender", ErrNilUserAccount)
+	}
+	if vmInput == nil {
+		return nil, ErrNilVmInput
+	}
+
+	senderAddr := acntSnd.AddressBytes()
+	senderIsNotCaller := !bytes.Equal(senderAddr, vmInput.CallerAddr)
```

Comment:
this cannot happen. blockChainHook gets the accounts according to vmInput. acntSnd == vmInput.Caller. always.

Reply:
yes, wanted to leave this as it is set in a different place, and should still hold.
If it cannot happen then all the better, this is harmless.
If something changes will be caught here and the function will not be executed.